Correctly define malloc_usable_size

Get correct abort() function definition for Solaris.

--- mozilla-esr31/memory/mozalloc/mozalloc.cpp.orig	2015-06-04 11:10:38.830585129 -0700
+++ mozilla-esr31/memory/mozalloc/mozalloc.cpp	2015-06-04 11:10:38.830522608 -0700
@@ -200,6 +200,8 @@
 }
 #endif // if defined(HAVE_VALLOC)
 
+extern "C" size_t malloc_usable_size(const void *ptr);
+
 size_t
 moz_malloc_usable_size(void *ptr)
 {
--- mozilla-esr31/memory/mozalloc/mozalloc_abort.cpp.orig	2015-06-04 11:10:38.830454359 -0700
+++ mozilla-esr31/memory/mozalloc/mozalloc_abort.cpp	2015-06-04 11:10:38.830381114 -0700
@@ -34,7 +34,11 @@
 // Define abort() here, so that it is used instead of the system abort(). This
 // lets us control the behavior when aborting, in order to get better results
 // on *NIX platforms. See mozalloc_abort for details.
+#if defined(SOLARIS)
+void std::abort(void)
+#else
 void abort(void)
+#endif
 {
     mozalloc_abort("Redirecting call to abort() to mozalloc_abort\n");
 }
